syntax = "proto3";

package pslx;

// the next will be 4
message Credentials {
    string user_name = 1;
    string password = 2;
    map<string, string> others = 3;
}

// the next will be 4
message FileInfo {
    string file_path = 1;
    int32 file_size = 2;
    int32 modified_time = 3;
}

// the next will be 7
message FrontendConfig {
    message ServerConfig {
        string server_url = 1;
    }

    ServerConfig container_backend_config = 2;
    ServerConfig instant_messaging_config = 3;
    ServerConfig email_config = 4;

    Credentials credential = 5;
    string galaxy_viewer_url = 6;
}
